cmd/kubebuilder/initproject/init.go

```diff
@@ -42,12 +42,14 @@ kubebuilder init repo --domain mydomain
 var domain string
 var copyright string
 var bazel bool
+var controllerOnly bool
 
 func AddInit(cmd *cobra.Command) {
 	cmd.AddCommand(repoCmd)
 	repoCmd.Flags().StringVar(&domain, "domain", "", "domain for the API groups")
 	repoCmd.Flags().StringVar(&copyright, "copyright", filepath.Join("hack", "boilerplate.go.txt"), "Location of copyright boilerplate file.")
 	repoCmd.Flags().BoolVar(&bazel, "bazel", false, "if true, setup Bazel workspace artifacts")
+	repoCmd.Flags().BoolVar(&controllerOnly, "controller-only", false, "if true, setup controller only")
```

Comment:
Can I add API resource to a project which has been initialized as "controller-only" ?

If yes, then does this flag makes sense ?

Reply:
It won't work since the `pkg/inject` won't build

Reply:
Hmm.... it will be great if we can remove this restriction.
Ideal workflow is - to be able to not to take this flag as input for `init` command and then be able to add resources for non-core type and controllers for core-types.

Lets discuss today to explore if we can make this happen.

Reply:
I agree with you, it's nice to remove the restriction. 
The divergence comes from the different dependencies for core type controller and CRD behaviors. Having a flag in `init` command, we can easily distinguish the different dependencies. 
Let's discuss to see if there's better solution.

Reply:
As discussed offline, we keep this flag for now. Later when the dynamic client is added in the kube builder library, we will try to remove this flag.